Common/Interfaces/IStreamReader.cs

```diff
@@ -34,8 +34,13 @@ public interface IStreamReader : IDisposable
         bool EndOfStream { get; }
 
         /// <summary>
-        /// Gets the next line/batch of content from the stream 
+        /// Gets the next line/batch of content from the stream
         /// </summary>
         string ReadLine();
+
+        /// <summary>
+        /// Gets whether or not this stream reader should be rate limited
+        /// </summary>
+        bool RateLimit { get; }
```

Comment:
Just my personal preference, this property name sounds better as `ShouldBeRateLimited`

Reply:
Your right 😄 I'll rename it